internal/pkg/template/diff/iter.go

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import "gopkg.in/yaml.v3"
+
+type action int
+
+const (
+	match        action = iota
+	modification        = iota
+	deletion            = iota
+	insertion           = iota
+	done                = iota
```

Comment:
```suggestion
	match        action = iota
	modification
	deletion
	insertion
	done
```

Reply:
nit: can we prefix the actions with `action` e.g.
```
actionMatch action = iota
actionMod
actionDel
acitionInsert
actionDone
```

Your code reminded me of this talk: https://go.dev/talks/2011/lex.slide#10